src/code/padmgr.c

```diff
@@ -298,53 +399,64 @@ void PadMgr_HandleRetraceMsg(PadMgr* padMgr) {
     padMgr->validCtrlrsMask = mask;
 
     if (gFaultMgr.msgId != 0) {
+        // If fault is active, no rumble
         PadMgr_RumbleStop(padMgr);
-    } else if (padMgr->rumbleOffFrames > 0) {
-        --padMgr->rumbleOffFrames;
+    } else if (padMgr->rumbleOffTimer > 0) {
+        // If the rumble off timer is active, no rumble
+        --padMgr->rumbleOffTimer;
         PadMgr_RumbleStop(padMgr);
-    } else if (padMgr->rumbleOnFrames == 0) {
+    } else if (padMgr->rumbleOnTimer == 0) {
+        // If the rumble on timer is inactive, no rumble
         PadMgr_RumbleStop(padMgr);
-    } else if (!padMgr->preNMIShutdown) {
-        PadMgr_RumbleControl(padMgr);
-        --padMgr->rumbleOnFrames;
+    } else if (!padMgr->isResetting) {
+        // If not resetting, update rumble
+        PadMgr_UpdateRumble(padMgr);
+        --padMgr->rumbleOnTimer;
     }
 }
 
 void PadMgr_HandlePreNMI(PadMgr* padMgr) {
     osSyncPrintf("padmgr_HandlePreNMI()\n");
-    padMgr->preNMIShutdown = true;
+    padMgr->isResetting = true;
     PadMgr_RumbleReset(padMgr);
 }
 
-void PadMgr_RequestPadData(PadMgr* padMgr, Input* inputs, s32 mode) {
+/**
+ * Fetches the most recently polled inputs from padmgr
+ *
+ * @param[out] inputs   Array of Input to copy inputs into
```

Comment:
Iirc last time [out] (and [in]) were discussed, the consensus was to not use them, so we should probably omit it here as well
```suggestion
 * @param inputs   Array of Input to copy inputs into
```